solr/docker/templates/Dockerfile.official.header.template

```diff
@@ -34,13 +34,13 @@ ARG SOLR_DOWNLOAD_URL
 # TODO: see patch in SOLR-15250 for some example ideas on fixing this to be more strict
 
 # Override the default solr download location with a prefered mirror, e.g.:
-#   docker build -t mine --build-arg SOLR_DOWNLOAD_SERVER=http://www-eu.apache.org/dist/lucene/solr .
+#   docker build -t mine --build-arg SOLR_DOWNLOAD_SERVER=http://www-eu.apache.org/dist/solr/solr .
```

Comment:
www-eu redirects to downloads:
```suggestion
#   docker build -t mine --build-arg SOLR_DOWNLOAD_SERVER=https://downloads.apache.org/solr/solr .
```

Have not thought closely about internal structure, but since we have solr-operator, it may make sense to add the extra `solr` level i.e.  `.../solr/solr/x.y.z/`.

Reply:
Ok cool. Yeah it looked like the structure was already going to be `.../solr/solr/x.y.z.`, which is fine with me, especially with the solr-operator stuff there (and other sub-projects in the future maybe?)